tests/scalers/couchdb/couchdb_test.go

```diff
@@ -0,0 +1,394 @@
+  minReplicaCount: 1
+  maxReplicaCount: 2
+  triggers:
+  - type: couchdb
+    metadata:
+      host: {{.HostName}}
+      port: "5984"
+      dbName: "animals"
+      queryValue: "1"
+      query: '{ "selector": { "feet": { "$gt": 0 } }, "fields": ["_id", "feet", "greeting"] }'
+      activationQueryValue: "1"
+      metricName: "global-metric"
+    authenticationRef:
+      name: {{.TriggerAuthName}}
+`
+)
+
+func CreateKubernetesResourcesCouchDB(t *testing.T, kc *kubernetes.Clientset, nsName string, data interface{}, templates []Template) {
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+}
+
+func TestCouchDBScaler(t *testing.T) {
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	// setup couchdb
+	CreateNamespace(t, kc, testNamespace)
+	installCouchDB(t)
+	
+	data, templates := getTemplateData(kc)
+	CreateKubernetesResourcesCouchDB(t, kc, testNamespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 3),
+		"replica count should be %d after 3 minutes", minReplicaCount)
+
+	// test scaling
+	testActivation(t, kc)
+	testScaleUp(t, kc)
+	testScaleDown(t, kc)
+
+	// cleanup
+	DeleteKubernetesResources(t, kc, testNamespace, data, templates)
+}
+
+func generateUUID() string {
+	id := strings.ReplaceAll(uuid.New().String(), "-", "")
+	return id
+}
+
+func installCouchDB(t *testing.T) {
+	_, err := ExecuteCommand(fmt.Sprintf("helm repo add couchdb %s", couchdbHelmRepo))
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+	_, err = ExecuteCommand("sudo helm repo update")
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+	uuid := generateUUID()
+	_, err = ExecuteCommand(fmt.Sprintf("sudo helm install test-release  --set couchdbConfig.couchdb.uuid=%s --namespace %s couchdb/couchdb", uuid, testNamespace))
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+}
+
+func getPassword(kc *kubernetes.Clientset, namespace string) string {
+	secret, _ := kc.CoreV1().Secrets(namespace).Get(context.Background(), "test-release-couchdb", metav1.GetOptions{})
+	encodedPassword := secret.Data["adminPassword"]
+	password := string(encodedPassword)
+	fmt.Println("Secret Data: ", password)
+	return password
+}
+
+func WaitForDeploymentReplicaCountChangeCouchDB(t *testing.T, kc *kubernetes.Clientset, name, namespace string, iterations, intervalSeconds int) bool {
+	t.Log("Waiting for some time to see if deployment replica count changes")
+	var replicas, prevReplicas int32
+	prevReplicas = -1
+
+	for i := 0; i < iterations; i++ {
+		deployment, _ := kc.AppsV1().Deployments(namespace).Get(context.Background(), name, metav1.GetOptions{})
+		replicas = deployment.Status.Replicas
+
+		t.Logf("Deployment - %s, Current  - %d", name, replicas)
+
+		if replicas != prevReplicas && prevReplicas != -1 {
+			break
+		}
+
+		prevReplicas = replicas
+		time.Sleep(time.Duration(intervalSeconds) * time.Second)
+	}
+
+	return int(replicas) > 1
+}
```

Comment:
There is a helper that you should use to wait for replicas

Reply:
The conditions are different for return statement, hence I made a different function